nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBCheck.java

```diff
@@ -0,0 +1,55 @@
+package org.nuxeo.ecm.core.storage.mongodb;
+
+import java.util.concurrent.TimeUnit;
+
+import org.bson.Document;
+import org.nuxeo.launcher.config.ConfigurationException;
+import org.nuxeo.launcher.config.ConfigurationGenerator;
+import org.nuxeo.launcher.config.backingservices.BackingChecker;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoClientURI;
+import com.mongodb.MongoTimeoutException;
+import com.mongodb.ServerAddress;
+
+public class MongoDBCheck implements BackingChecker {
+
+    private static final String TEMPLATE_NAME = "mongodb";
+
+    @Override
+    public boolean acceptConfiguration(ConfigurationGenerator cg) {
+        return cg.getTemplateList().contains(TEMPLATE_NAME);
+
+    }
+
+    @Override
+    public void check(ConfigurationGenerator cg) throws ConfigurationException {
+        MongoClient ret = null;
+        String serverName = cg.getUserConfig().getProperty(ConfigurationGenerator.PARAM_MONGODB_SERVER);
+        String dbName = cg.getUserConfig().getProperty(ConfigurationGenerator.PARAM_MONGODB_NAME);
+
+        MongoClientOptions.Builder optionsBuilder = MongoClientOptions.builder()
```

Comment:
I tried to did that but this ended up as being not a good idea for two reasons :

- The connection parameters are not the same
- Factored code pulls dependencies that are not in the classpath at that moment (for instance `MongoDBRepository` -> `MongoDBRepsitoryDescriptor` -> `DBSRepositoryDescriptor`